rllib/optimizers/sync_replay_optimizer.py

```diff
@@ -156,7 +163,13 @@ def stats(self):
             })
 
     def _optimize(self):
-        samples = self._replay()
+        if self._fake_batch:
+            fake_batch = SampleBatch(self._fake_batch)
+            samples = MultiAgentBatch({
+                DEFAULT_POLICY_ID: fake_batch
+            }, fake_batch.count)
+        else:
+            samples = self._replay()
```

Comment:
Could also monkey patch self._replay to return the fake batch in the test, instead of adding code explicitly here in the optimizer.

Reply:
+1, much better!